### PR TITLE
HW2 Task05

### PR DESCRIPTION
**Pull request by LinaKrevetochka**

### CODE REVIEW

File: Homework2/Task05/task05.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import string
+assert = custom_range(string.ascii_lowercase, 'g') == ['a', 'b', 'c', 'd', 'e', 'f']
+assert = custom_range(string.ascii_lowercase, 'g', 'p') == ['g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o']
+assert = custom_range(string.ascii_lowercase, 'p', 'g', -2) == ['p', 'n', 'l', 'j', 'h']
+"""
+from typing import List, Sequence
+
+
+def custom_range(
+    seq: Sequence, start: str, stop: str = None, step: str = None
```
Comment (Cjkjvfnby) [inline Homework2/Task05/task05.py]: Why all of them `str`?

File: Homework2/Task05/task05.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+"""
+Some of the functions have a bit cumbersome behavior when we deal with
+positional and keyword arguments.
+Write a function that accept any iterable of unique values and then
```
Comment (Cjkjvfnby) [inline Homework2/Task05/task05.py:4]: `any iterable of unique values`  what does it mean?

File: Homework2/Task05/task05.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    res = []
+    while True:
+        if step > 0 and seq.index(start) >= seq.index(stop):
+            break
+        elif step < 0 and seq.index(start) <= seq.index(stop):
+            break
+        else:
+            ind = seq.index(start)
+            res.append(seq[ind])
+            start = seq[ind + step]
```
Comment (Cjkjvfnby) [inline Homework2/Task05/task05.py]: You play with the indexes. What prevents you form the Index errors?   Could it be the absence of the tests?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: Homework2/Task05/task05.py (hunk 3 above)
Comment (LinaKrevetochka) [reply Homework2/Task05/task05.py]: My bad. Should it give an exeption or I can return empty list?

Comment (Cjkjvfnby) [reply Homework2/Task05/task05.py]: These kinds of functions are usually errorless.  ('ab', 'a', 'b', 100)  - > 'a'

File: Homework2/Task05/task05.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+it behaves as range function:
+import string
+assert = custom_range(string.ascii_lowercase, 'g') == ['a', 'b', 'c', 'd', 'e', 'f']
+assert = custom_range(string.ascii_lowercase, 'g', 'p') == ['g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o']
+assert = custom_range(string.ascii_lowercase, 'p', 'g', -2) == ['p', 'n', 'l', 'j', 'h']
+"""
+from typing import Any, List, Sequence
+
+
+def check_side_to_go(ind_start: int, ind_stop: int, step: int) -> bool:
```
Comment (Cjkjvfnby) [inline Homework2/Task05/task05.py:14]: Imagine that diaalog:
```
- What side should I go? 
- True
```
I'd say `check_side_to_go` is not the best name for that function.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**